autoPyTorch/pipeline/components/training/trainer/utils.py

```diff
@@ -0,0 +1,165 @@
+import re
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.hyperparameters import (
+    UniformFloatHyperparameter,
+    UniformIntegerHyperparameter
+)
+
+import torch
+from torch.optim.optimizer import Optimizer
+
+
+def update_model_state_dict_from_swa(model: torch.nn.Module, swa_state_dict: Dict) -> None:
+    """
+    swa model adds a module keyword to each parameter,
+    this function updates the state dict of the model
+    using the state dict of the swa model
+    Args:
+        model:
+        swa_state_dict:
+
+    Returns:
+
+    """
+    model_state = model.state_dict()
+    for name, param in swa_state_dict.items():
+        name = re.sub('module.', '', name)
+        if name not in model_state.keys():
+            continue
+        model_state[name].copy_(param)
+
+
+class Lookahead(Optimizer):
+    r"""PyTorch implementation of the lookahead wrapper.
+    Lookahead Optimizer: https://arxiv.org/abs/1907.08610
+    """
+
+    def __init__(self, optimizer: Optimizer, config: Dict[str, Any]) -> None:
+        """optimizer: inner optimizer
+        la_steps (int): number of lookahead steps
+        la_alpha (float): linear interpolation factor. 1.0 recovers the inner optimizer.
+        pullback_momentum (str): change to inner optimizer momentum on interpolation update
```

Comment:
I think we no longer have this option and later below is hardcoded?

Reply:
i'll remove it from the doc